app/Libraries/Search/BeatmapsetSearch.php

```diff
@@ -111,6 +112,19 @@ private function addModeFilter($query)
         }
     }
 
+    private function addPlayedFilter($query)
+    {
+        if ($this->params->user === null || !$this->params->user->isSupporter()) {
+            return;
+        }
+
+        if ($this->params->playedState === PlayedState::PLAYED) {
+            $query->filter(['terms' => ['difficulties.beatmap_id' => $this->getPlayedBeatmapIds()]]);
```

Comment:
seems to be slightly faster just filtering by `beatmap_id` instead of running the extra query to get the `beatmapset_ids`, so we should probably make another PR to update the other one